simulation/traffic_simulator/include/traffic_simulator/simulation_clock/simulation_clock.hpp

```diff
@@ -29,18 +29,34 @@ class SimulationClock : rclcpp::Clock
 
   auto getCurrentRosTimeAsMsg() -> rosgraph_msgs::msg::Clock;
 
-  auto getCurrentScenarioTime() const { return (frame_ - frame_offset_) / frame_rate; }
+  auto getCurrentScenarioTime() const
+  {
+    auto nanoseconds = nanoseconds_count - nanoseconds_offset;
```

Comment:
As we talked about, please simplify the code.

Reply:
Changed to storing time in seconds as a single double.